app-react/src/main/kotlin/com/ustadmobile/view/CourseDiscussionDetailScreen.kt

```diff
@@ -0,0 +1,182 @@
+import mui.icons.material.*
+import mui.material.*
+import mui.material.List
+import mui.material.styles.TypographyVariant
+import mui.system.responsive
+import mui.system.sx
+import react.FC
+import react.Props
+import react.create
+import react.useMemo
+import kotlin.js.Date
+
+external interface CourseDiscussionDetailProps: Props {
+    var uiState: CourseDiscussionDetailUiState
+    var onClickPost: (DiscussionPostWithDetails) -> Unit
+    var onDeleteClick: (DiscussionPostWithDetails) -> Unit
+    var onClickAddItem: () -> Unit
+}
+
+val CourseDiscussionDetailComponent2 = FC<CourseDiscussionDetailProps> { props ->
+
+    val strings = useStringsXml()
+
+
+
+    Container {
+
+        Typography {
+            + props.uiState.courseDiscussion?.courseDiscussionTitle.toString()
+        }
+
+        Box{
+            sx {
+                height = 10.px
+            }
+        }
+
+        Typography {
+            variant = TypographyVariant.caption
+            + strings[MessageID.description]
+        }
+
+        Typography {
+            + props.uiState.courseDiscussion?.courseDiscussionDesc.toString()
+        }
+
+
+        maxWidth = "lg"
+
+        Stack {
+            direction = responsive(mui.material.StackDirection.column)
+            spacing = responsive(10.px)
+
+
+
+
+        }
+
+        Box{
+            sx {
+                height = 10.px
+            }
+        }
+
+        Typography {
+            variant = TypographyVariant.h6
+            + strings[MessageID.posts]
+        }
+
+        List {
+            //Add post:
+            ListItem {
+                disablePadding = true
+
+                ListItemButton {
+                    onClick = {
+                        props.onClickAddItem()
+                    }
+
+                    ListItemIcon {
+                        Add {}
+                    }
+
+                    ListItemText {
+                        +(strings[MessageID.post])
+                    }
+                }
+            }
+
+            //Posts:
+            props.uiState.posts.forEach { item ->
+                ListItem {
+                    disablePadding = true
+//                    secondaryAction = IconButton.create {
+//                        onClick = {
+//                            props.onDeleteClick(item)
+//                        }
+//                        Delete {}
+//                    }
+
+                    ListItemButton {
+                        ListItemIcon {
+                            UstadBlankIcon { }
+                        }
+
+                        onClick = {
+                            props.onClickPost(item)
+                        }
+
+                        val dateFormatted = useMemo(dependencies = arrayOf(item.postLatestMessageTimestamp)) {
+                            Date(item.postLatestMessageTimestamp ?: 0L).toLocaleDateString()
+                        }
+
+                        UstadMessageField{
```

Comment:
We shouldn't use this inside a ListItem. I think we should be using ListItem as is. The formatting and alignment goes wrong because there is a blank icon in the ListItem itself, and then an icon in the content.